gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiResource.java

```diff
@@ -851,11 +852,17 @@ private void setPicturesUrl(final GenericApiEntity apiEntity) {
         String backgroundUrl = uriBuilder.build().toString();
 
         if (apiEntity.getDefinitionVersion() == DefinitionVersion.V4) {
-            ApiEntity apiEntityV4 = (ApiEntity) apiEntity;
-            apiEntityV4.setPictureUrl(pictureUrl);
-            apiEntityV4.setPicture(null);
-            apiEntityV4.setBackgroundUrl(backgroundUrl);
-            apiEntityV4.setBackground(null);
+            if (apiEntity instanceof ApiEntity apiEntityV4) {
+                apiEntityV4.setPictureUrl(pictureUrl);
+                apiEntityV4.setPicture(null);
+                apiEntityV4.setBackgroundUrl(backgroundUrl);
+                apiEntityV4.setBackground(null);
+            } else if (apiEntity instanceof NativeApiEntity nativeApiEntityV4) {
+                nativeApiEntityV4.setPictureUrl(pictureUrl);
+                nativeApiEntityV4.setPicture(null);
+                nativeApiEntityV4.setBackgroundUrl(backgroundUrl);
+                nativeApiEntityV4.setBackground(null);
+            }
```

Comment:
picture and background are not part of the generic Entity ?

Reply:
The getters for picture, picture url, background and background url are part of the generic api entity, but not the setters 🙈

Reply:
I have added setters for picture + background in the `GenericApiEntity`. Seeing as `pictureUrl` and `backgroundUrl` are at the inheritor level, I am still forced to cast to either a `ApiEntity` or `NativeApiEntity` to set those fields.